### PR TITLE
feat: create permission resource

### PR DESCRIPTION
**Pull request by tdien105**

### CODE REVIEW

File: internal/provider/mssql_permission_resource.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+			"principal_id": schema.StringAttribute{
+				MarkdownDescription: "`<principal_id>`.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"permissions": schema.StringAttribute{
+				MarkdownDescription: "Name of database-level SQL permission. For full list of supported permissions, see [docs](https://learn.microsoft.com/en-us/sql/t-sql/statements/grant-database-permissions-transact-sql?view=azuresqldb-current#remarks)",
+				Computed:            true,
```
Comment (locmai) [inline internal/provider/mssql_permission_resource.go:63]:
```suggestion
			"permission": schema.StringAttribute{
				MarkdownDescription: "Name of database-level SQL permission. For full list of supported permissions, see [docs](https://learn.microsoft.com/en-us/sql/t-sql/statements/grant-database-permissions-transact-sql?view=azuresqldb-current#remarks)",
				Required:            true,
```

Either updating the MssqlPermissionResourceModel struct to ``Permissions []types.String `tfsdk:"permissions"` `` or this to ``permission``. The Computed will set this to be read-only data.

File: internal/mssql/sqlclient.go [hunk 2]
```diff
@@ -331,3 +331,65 @@ func (m client) UnassignRole(ctx context.Context, role string, principal string)
...
+func (m client) GrantDatabasePermission(ctx context.Context, principal_id string, permission_name string) (DatabasePermission, error) {
+	var DatabasePermission DatabasePermission
+
+	conn, err := m.connect(nil)
+	if err != nil {
+		return DatabasePermission, err
+	}
+	defer conn.Close()
+
+	cmd := fmt.Sprintf("GRANT %s TO [%s]", permission_name, principal_id)
```
Comment (locmai) [inline internal/mssql/sqlclient.go:371]: This got a syntax error when executed:

```
mssql: Incorrect syntax near the keyword '<permissions>'.
```

**Review by locmai — CHANGES_REQUESTED** [review]
Besides the gofmt errors

**Comment by vsabella** [comment]
integrating into #9

**Pull request closed without merging**